### ISSUE TITLE
openURL method does not override method from its superclass

### ISSUE THREAD

**Issue by MrCharli3** [TomGrill/gdx-facebook#46]
It seems the openURL-method does not exist anymore? I am getting a "method does not override method from its superclass". Has it been replaced by something else?

FBSDKAppEvents openURL expects a NSPropertyList, but roboVM only provides a method with a NSObject, which must be something new since it used to work. 

Removing it makes the facebook login stop working, so I am assuming I should replace it with something new?

To be clear, this method causes the error:
```
@Override
    public boolean openURL(UIApplication application, NSURL url, String sourceApplication, NSPropertyList annotation) {
        // You need to add this line, otherwise Facebook will not work
        // correctly!
        return FBSDKApplicationDelegate.getSharedInstance().openURL(application, url, sourceApplication, annotation);

    }
```

**Comment by TomGrill** [comment]
Which libgdx version?

**Comment by MrCharli3** [comment]
LibGdx: 1.9.5
RoboVm: 2.3.0
GdxFacebook: 1.2.4

Here is my IOSLauncher:

```
import org.robovm.apple.foundation.NSAutoreleasePool;
import org.robovm.apple.foundation.NSPropertyList;
import org.robovm.apple.foundation.NSURL;
import org.robovm.apple.uikit.UIApplication;
import org.robovm.apple.uikit.UIApplicationLaunchOptions;
import org.robovm.pods.facebook.core.FBSDKAppEvents;
import org.robovm.pods.facebook.core.FBSDKApplicationDelegate;

import com.badlogic.gdx.backends.iosrobovm.IOSApplication;
import com.badlogic.gdx.backends.iosrobovm.IOSApplicationConfiguration;
import com.my.game.MyGdxGame;

public class IOSLauncher extends IOSApplication.Delegate {

    MyGdxGame game;

    @Override
    protected IOSApplication createApplication() {
        IOSApplicationConfiguration config = new IOSApplicationConfiguration();
        config.orientationLandscape = false;
        config.orientationPortrait = true;
        config.useCompass = false;
        config.useAccelerometer = false;
        game = new MyGdxGame(new IOSPlatform());
        return new IOSApplication(game, config);
    }

    public static void main(String[] argv) {
        NSAutoreleasePool pool = new NSAutoreleasePool();
        UIApplication.main(argv, null, IOSLauncher.class);
        pool.close();
    }

    @Override
    public void didBecomeActive(UIApplication application) {
        super.didBecomeActive(application);
        // You need to add this line, otherwise Facebook will not work
        // correctly!
        FBSDKAppEvents.activateApp();
    }
    
    @Override //this does not override any method
    public boolean openURL(UIApplication application, NSURL url, String sourceApplication, NSPropertyList annotation) {
        // You need to add this line, otherwise Facebook will not work
        // correctly!
        return FBSDKApplicationDelegate.getSharedInstance().openURL(application, url, sourceApplication, annotation);

    }

    @Override
    public boolean didFinishLaunching(UIApplication application, UIApplicationLaunchOptions launchOptions) {
        boolean finished = super.didFinishLaunching(application, launchOptions);
        FBSDKApplicationDelegate.getSharedInstance().didFinishLaunching(application, launchOptions);
        game.setPlatformResolver(new IOSResolver(game));
        return finished;
    }

}
```

(I wan't aware you were active on libgdx forum, maybe better to answer here? I can forward the solution (if any) to the forum later)

**Comment by TomGrill** [comment]
Please try to pass null as NSObject. Like this: FBSDKApplicationDelegate.getSharedInstance().openURL(application, url, sourceApplication, null); and tell me if that works. Cannot test right now.

**Comment by MrCharli3** [comment]
that solved it, not pretty though ^^ The method from RoboVm is deprecated as well. Thanks for your help and for a great library! :)

**Comment by TomGrill** [comment]
True, but no other option available at the moment :)

**Comment by kali-gandaki** [comment]
Hi MrCharli3 , 

Can you please specifiy the exact method you called ? My openUrl method looks like this .
 @Override
    public boolean openURL(UIApplication application, NSURL url, String sourceApplication, NSObject annotation) {
        //return super.openURL(application, url, sourceApplication, annotation);
        return FBSDKApplicationDelegate.getSharedInstance().openURL(application,url,sourceApplication,null);
    }
 For me , the facebook login dialog pops up but i get the following error up login  , login in not successful .

[debug] gdx-facebook (1.2.4): Could not load existing accessToken.
2017-03-08 11:00:22.515 XXX[968:17076] [debug] gdx-facebook (1.2.4): Starting GUI sign in.
2017-03-08 11:00:22.528 XXX[968:17076] -canOpenURL: failed for URL: "fbauth2:/" - error: "The operation couldn’t be completed. (OSStatus error -10814.)"
2017-03-08 11:00:22.534 XXX[968:17076] -canOpenURL: failed for URL: "fbauth2:/" - error: "The operation couldn’t be completed. (OSStatus error -10814.)"
2017-03-08 11:01:46.031 XXX[968:17076] [debug] gdx-facebook (1.2.4): Sign fail by user.
2017-03-08 11:01:46.031 XXX[968:17076] [info] FacebookService: Login cancelled

**Comment by MrCharli3** [comment]
@kali-gandaki 
My method looks the exact same. Your error is that you cannot load an existing accessToken, which sounds odd that it would even try that. Do you check somewhere in your code if you are already signed in? that might trigger an error like this I can imagine. 

Also, what versions are you using?

**Comment by kali-gandaki** [comment]
Hi I am using libgdx 1.9.5 and robovm 2.3.0 and using 1.2.5-snapshot version of gdx-facebook .
It works perfectly with android and desktop but not in ios . Everything seems to work ( login dialog , login process , app authorization ) but in the result I get sign in failed and login cancelled . 

@TomGrill Can you help me regarding this ?

**Comment by TomGrill** [comment]
Ok this problem seems to be a huge issue. I will look for a solution this weekend.

**Comment by TomGrill** [comment]
Fixed with version 1.3.0. Please update.

**Comment by kali-gandaki** [comment]
Hi . Thanks its working now .
